deephyper/search/nas/model/train_utils.py

```diff
@@ -69,12 +69,53 @@ def acc(y_true, y_pred):
     return tf.keras.metrics.categorical_accuracy(y_true, y_pred)
 
 
+# BRUCEEDIT
+def por2(y_true, y_pred):
+    zero = tf.constant(0, dtype=tf.float32)
+    where = tf.not_equal(y_true, zero)
+    total_error = tf.reduce_sum(tf.square(tf.subtract(tf.boolean_mask(y_true, where),
+                                                      tf.reduce_mean(tf.boolean_mask(y_true, where)))))
+    unexplained_error = tf.reduce_sum(tf.square(tf.subtract(tf.boolean_mask(y_true, where),
+                                                            tf.boolean_mask(y_pred, where))))
+    Prot_R_squared = tf.subtract(1.0, tf.div(unexplained_error, total_error))
+
+    return Prot_R_squared
+
+
+def po(y_true, y_pred):
+    y_true = tf.math.abs(y_true)
+    num_non_zero = tf.cast(tf.math.count_nonzero(y_true), tf.float32)
+    custom_loss = tf.div(tf.reduce_mean(tf.squared_difference(y_true, y_pred)), num_non_zero)
+    return -custom_loss
+
+
+def poloss(y_true, y_pred):
+    y_true = tf.math.abs(y_true)
+    num_non_zero = tf.cast(tf.math.count_nonzero(y_true), tf.float32)
+    custom_loss = tf.div(tf.reduce_mean(tf.squared_difference(y_true, y_pred)), num_non_zero)
+    return custom_loss
+
+
+def negmae(y_true, y_pred):
+    return -tf.keras.metrics.mean_absolute_error(y_true, y_pred)
+
+def negmse(y_true, y_pred):
+    return -tf.keras.metrics.mean_squared_error(y_true, y_pred)
+
+
 metrics = OrderedDict()
 metrics["mean_absolute_error"] = metrics["mae"] = mae
 metrics["r2"] = r2
 metrics["mean_squared_error"] = metrics["mse"] = mse
 metrics["accuracy"] = metrics["acc"] = acc
 
+#BRUCEEDIT
```

Comment:
Remove this comment you can just simply add them because the version controller `git` is already tracking who is making the change.